ip-allow-listing-route-service/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/http/httputil"
+	"net/netip"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var (
+	ErrBadRequest                   = fmt.Errorf("bad request")
+	ErrForbidden                    = fmt.Errorf("forbidden")
+	ErrMissingOrInvalidForwardedURL = fmt.Errorf("%w: invalid or missing x-cf-forwarded-url header", ErrBadRequest)
+	ErrMissingTrueClientIP          = fmt.Errorf("%w: missing x-cf-true-client-ip header", ErrBadRequest)
+	ErrInvalidTrueClientIP          = fmt.Errorf("%w: invalid x-cf-true-client-ip header", ErrBadRequest)
+)
+
+func main() {
+	slog.SetDefault(slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelDebug})))
+
+	err := Main()
+	if err != nil {
+		slog.Error("route-service failed", "error", err)
+		os.Exit(1)
+	}
+}
+
+func Main() error {
+	allowListFile := os.Getenv("ALLOW_LIST_FILE")
+	if allowListFile == "" {
+		return fmt.Errorf("no file to read prefixes from provided")
+	}
+
+	allowedPrefixes, err := loadPrefixes(allowListFile)
+	if err != nil {
+		return err
+	}
+
+	s := http.Server{
+		Addr: ":" + os.Getenv("PORT"),
+		Handler: &httputil.ReverseProxy{
+			Director:     proxyDirector,
+			ErrorHandler: proxyErrorHandler,
+			Transport: &transport{
+				allowedPrefixes: allowedPrefixes,
+				roundTripper:    &http.Transport{},
+			},
+		},
+	}
+
+	slog.Info("starting server", "allow-list", allowedPrefixes, "addr", s.Addr)
+
+	return s.ListenAndServe()
+}
+
+func loadPrefixes(path string) (prefixes []netip.Prefix, err error) {
```

Comment:
I know, Go `net.netip` package calls this "Prefix", but we also often talk about "CIDR" for the same thing.

Maybe we could call this `parseAllowedCIDRs`?

Reply:
I think people can be trusted to understand that those two terms are pretty much the same thing.

Reply:
The README mentions a list of `CIDR`s.

Reply:
Unified it to IP-prefix.